### ISSUE TITLE
resource name not getting cost analysis (Usageapi cllent)

### ISSUE THREAD

**Issue by Gopichand-Heeddata** [oracle/oci-python-sdk#637]
In the usage API call, many values return as "none". In the JSON response, the "resource name" key is present but its value is null. Could you assist me in making the correct API call to retrieve these values?  @adizohar, for your help

```
  "compartment_id": null,
  "compartment_name": null,
  "compartment_path": null,
  "computed_amount": null,
  "computed_quantity": 0.0,
  "currency": " ",
  "discount": null,
  "is_forecast": false,
  "list_rate": null,
  "overage": null,
  "overages_flag": null,
  "platform": null,
 **"resource_name": null,**
  "service": "Virtual Cloud Network",
  "shape": null,
 "tenant_id": null,
  "tenant_name": null,
  "time_usage_ended": "2023-05-14T00:00:00+00:00",
  "time_usage_started": "2023-05-01T00:00:00+00:00",
  "unit": null,
  **"unit_price": null,**
  "weight": null

```

**Comment by adizohar** [comment]
UsageAPI aggregate the data based on grouping, those fields only be populated if you group by them, please read the documentation for the usage_api

**Comment by adizohar** [comment]
If you want the resource_name, try to use group_by the resource_name

**Comment by adizohar** [comment]
Indeed, resource_name is not supported key, it is not in the documentation, it was added for future functionality

**Comment by adizohar** [comment]
https://docs.oracle.com/en-us/iaas/tools/python/2.125.1/api/usage_api/models/oci.usage_api.models.RequestSummarizedUsagesDetails.html

Currently Supported - “compartmentName”, “compartmentPath”, “compartmentId”, “platform”, “region”, “logicalAd”, “resourceId”, “tenantId”, “tenantName”]`

**Comment by Gopichand-Heeddata** [comment]
> Currently Supported - “compartmentName”, “compartmentPath”, “compartmentId”, “platform”, “region”, “logicalAd”, “resourceId”, “tenantId”, “tenantName”]`

Okay thankyou for confirming @adizohar, when we able to get resource_name in usageapi.
is there any way to  get the cost by resource name ?

**Comment by adizohar** [comment]
It was requested by many customers, metering team is working on it., I don't have any timeline

**Comment by Gopichand-Heeddata** [comment]
Hi, @adizohar ,

how can I get the resource name other than the usage API call is there any other API call available or any other search options available
if any other approaches are there, let me know.

**Comment by adizohar** [comment]
The easiest way is to use the search api:
```
#################################################################################
# search.py
# https://docs.oracle.com/en-us/iaas/Content/Search/Concepts/queryoverview.htm
#################################################################################
import oci

config_file = "~/.oci/config"
config = oci.config.from_file(config_file, "DEFAULT")
resource_ocid = "ocid1.instance.oc1.iad.xxxxxxxxxxx"
query = "query all resources where identifier == '" + resource_ocid + "'"

search_client = oci.resource_search.ResourceSearchClient(config)
reponse_data = search_client.search_resources(
    search_details=oci.resource_search.models.StructuredSearchDetails(query=query)
).data

name = reponse_data.items[0].display_name if reponse_data.items else ""
print(name)
```

**Comment by Gopichand-Heeddata** [comment]
Thanks @adizohar 

in the above api call I need to do an individual search for every resource I need. 
in this approach if i need 100 resources details, then it is not a better approach or if we need more resources like more than 100 like wise. 
is there any better approach.
what do you suggest?

**Comment by adizohar** [comment]
For now, Search is the best option
You can also run [showoci](https://github.com/oracle/oci-python-sdk/tree/master/examples/showoci)
Produce CSV files, there is one CSV all_resources.csv which has most of the resources

**Comment by alvenkatesh** [comment]
hi, @adizohar 

does we have any limitations to the api call we made using search api, like the number of api calls can be made to that particular tenancy.

as per my knowledge we can made only 1000 api calls for a particular tenancy. is it correct?

**Comment by adizohar** [comment]
I am not aware of any API limitation, if you call too quickly, the service limits you and the retry option will work
Please check - https://docs.oracle.com/en-us/iaas/tools/python/2.125.2/sdk_behaviors/retries.html
